sources/assets/Stride.Core.Assets/Reflection/CollectionItemIdentifiers.cs

```diff
@@ -134,7 +134,12 @@ public void Validate(bool isList)
         public object GetKey(ItemId itemId)
         {
             // TODO: add indexing by guid to avoid O(n)
-            return keyToIdMap.SingleOrDefault(x => x.Value == itemId).Key;
+            foreach (var kvp in keyToIdMap)
+            {
+                if (kvp.Value == itemId)
+                    return kvp.Key;
+            }
+            return null;
```

Comment:
Again why reinventing the wheel? That is the exact implementation of `SingleOrDefault`.

It does miss a null-check:

```csharp
return keyToIdMap.SingleOrDefault(x => x.Value == itemId)?.Key;
```

Reply:
Actually this is not quite right, single should throw if there's not exactly one match, I'll change this.